### DISCUSSION TITLE
Suggestions

### DISCUSSION THREAD

**Discussion by cloneofghosts** [Pirate-Weather/pirateweather#62] [Ideas]
Adding a GitHub project to the repository might help you better keep track of things though in order for something to properly show up on the project they need to be an issue or a pull request. There's even support for custom fields so you can add priority as an example.

I haven't used projects on GitHub much myself though it does seem very similar to Trello which I've used in the past. The documentation for projects can be found here [https://docs.github.com/en/issues/planning-and-tracking-with-projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects)

I thought of a couple more ideas that I'll add below:

**Social Media**
I know you have your own personal socials but having a dedicated PW account might be useful. Could post stuff about new versions, any maintenance, etc. I'm sure there's more but that was what I could think of off the top of my head.

**Blog**
I know you mentioned in another issue about writing up an analysis about the May 1st downtime and maybe having a dedicated blog page might be useful for this? I know other sites have used <https://substack.com> or even <https://medium.com> so it might be worth looking into to?

**Comment by cloneofghosts** [comment]
@alexander0042 Since Twitter seems to be imploding maybe having a PW Mastodon account instead? I don't know what people are going to be moving to but since you already have an account there it might make sense to create it there.

EDIT: What about the idea of creating issue templates for the repos? You could setup the following:

**HA Repo**
bug
enhancement
general

**API Repo**
bug
enhancement
apiable
general

You can set it up so it automatically adds labels to the issue and you could do it for bug/enhancement/apiable. I could also maybe setup a PR to do this?

**Reply by alexander0042** [reply]
LOVE the issue template idea! My first question is almost always "What location/ time is this for, what version is this, and what was the expected result". For HA, same thing, with the addition of what HA version they're using. 

For the socials, I think it's easier to just keep using my personal ones, since I have them setup already. However, the blog idea is a great one, and I actually have three posts in the works (I need to as part of my AWS community builders membership anyway). The topics so far are the downtime analysis, new Kong infrastructure, and a summary of the cost structure. I'm thinking I'll just keep it on the main readthedocs url, since it's already setup and works pretty well

**Reply by cloneofghosts** [reply]
**Issue Template** - I haven't used this much myself but GitHub has issue form templates in beta which is what you can use in order for people to fill out certain information before posting. Here's the documentation I found for it https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms. Also thinking it might be a good idea to automatically add Needs Review to items created through the template? Would be easier for you when checking through items to see which ones you haven't looked at yet.

**Socials** - Fair enough. I don't really see it advertised much anywhere (apart from your profile) and I don't see too many PW related posts but maybe it's just a case of using it more?

**Blog** - Again fair enough. Using the existing setup means there is one less website to maintain and might make things easier on your end.

**Comment by cloneofghosts** [comment]
@alexander0042 Was thinking of making a separate post somewhere for this but as mentioned here https://github.com/alexander0042/pirate-weather-ha/issues/30#issuecomment-1747280200 it would be nice to have either a github project for the 2.0 update or add a section on the roadmap.

With the new GitHub projects there is the ability to have separate repos share the same project though you'd want to turn off items being automatically added to the project, If you use a project it's easier to update and we can see how things are progressing without you having to update the roadmap file each time something is added/removed.

**Comment by cloneofghosts** [comment]
@alexander0042 I know you're busy but I definitely thing you either need someone to help you manage the GitHub repo or set something up to remind you to check in on the repo more often? From what I've noticed you'll be active for a bit and then go MIA for a few weeks at a time replying to a few issues.

**Reply by alexander0042** [reply]
You are right on target with this- my available time to work on this tends to ebb around, which is why you see the pattern you do. While I try to respond to any mission critical thing stat, I could do a much better job keeping up with day to day things. Couple action items then:

1. I've set up a new notification to check in at least once a week, which isn't exactly a breakneck pace, but should provide at least a baseline. 
2. Setting up a project now, since I think that will help a lot to keep V2.0 organized.
3. ... are you volunteering? 😊 No pressure, but I would happily add you in as an admin here to help with some of the management side of things?

**Reply by cloneofghosts** [reply]
I don't mind helping out if that would make things easier on your end.

**Reply by alexander0042** [reply]
Welcome to the club- if you send me your e-mail, I can also add you to the slack channel that's used for coordination between me and MerrySky 😊

I'm going through issues today and tomorrow, so should be able to tame that a bit shortly.

**Reply by cloneofghosts** [reply]
I went and cleaned up some issues and added the Needs Review tag to some newer ones which you've missed to hopefully help with some things.

**Reply by alexander0042** [reply]
That's perfect, thanks!

**Reply by cloneofghosts** [reply]
I can also help out with the HA repo if you would like as well.

I mentioned this in one of my issues but having a PW organization on GitHub with all the PW repositories might be useful. It might break existing links which would be annoying.

**Reply by cloneofghosts** [reply]
I see you've created an organization and added me to the HA repo.

Not sure which repos you plan on adding but here's a list of others you could add:

https://github.com/alexander0042/weather-vue (could be archived)
https://github.com/alexander0042/Pirate-Weather-SMSL
https://github.com/alexander0042/alexander0042.github.io (could be archived)
https://github.com/alexander0042/translations

**Reply by alexander0042** [reply]
Yes! Making an organization was a great suggestion, and I can see how it will really keep things contained. I did some testing and I don't think it should break too many links, since GitHub forwards everything, but I'm keeping an eye out for that. I'm going to archive [alexander0042.github.io](https://github.com/alexander0042/alexander0042.github.io), and move the rest of them over.

**Reply by cloneofghosts** [reply]
I noticed you created a project for the 2.0 update. I linked it up to the HA repo so you can add items from that repo into there. I added a few items that are related to items already in the repo and added a priority and est completion date field which you can use if you wish.

Could always create another project for the items not included in the 2.0 update for people to keep track of?

**Reply by cloneofghosts** [reply]
I've created two more projects that might be helpful for you to keep track of things. I can't make them public unfortunately but the links are here

https://github.com/orgs/Pirate-Weather/projects/3 (HA)
https://github.com/orgs/Pirate-Weather/projects/4 (non 2.0 API items)

**Reply by alexander0042** [reply]
Thanks for making these! I really like this format, and also switched them to public. I'm taking a look at them now to see if there are any other changes/ adjustments that would make sense

**Reply by cloneofghosts** [reply]
Feel free to move things around as needed. May be a good idea to have the 2.0 and the api project linked on the road map page?

**Comment by cloneofghosts** [comment]
@alexander0042 I know it's the summer and you're likely busier than usual but I've noticed you've been MIA more than usual. I've been trying to help solve issues where I can but there's only so much I can help with before I need to send it off to you to look at.

**Reply by alexander0042** [reply]
I’m really sorry about this, and trust me that I notice it as well! I can’t say enough how much I appreciate you running the initial triage, and you are absolutely doing an amazing job with it. I totally see how it’s frustrating that there’s things that need me to do, and so I don’t take it lightly. 

All I’ve got to add is to say that the worst of my scheduling issues are past for now, and so things should start rolling again! I really appreciate the tagging and assigning things, since that gives me a priority order when I sift though then.

**Reply by cloneofghosts** [reply]
No worries. I figured you were busy and with it being summer you're trying to get out an enjoy the weather while you can. Anything assigned to you or tagged as Needs Review are things that you need to look at though I'd focus on releasing V2.1 first.

I've also started using milestones to give a rough idea on what API version the issue will be added/fixed in. So the 2.1 milestone contains stuff which will release is 2.1 and 2.1.x just indicates that it will be added to the API in a patch to version 2.1.

**Comment by cloneofghosts** [comment]
@alexander0042 Just out of curiosity you still have the reminder to check the GitHub issues? I noticed that outside of urgent issues you sometimes go a week or two without replying to any issues.

**Reply by cloneofghosts** [reply]
@alexander0042 I tagged items in this repo and the HA repo as awaiting developer response for things that we are waiting on you for. Whenever you  get the chance to look at them it'd be great. :smile:

**Reply by alexander0042** [reply]
So sorry about that! What happens is that I get focused on something I think is almost done (in this case self-hosting), and think "I'll just reply to the other stuff once that's done". However, when it ends up taking longer than I anticipate, it creates these major issues. On it now!

**Reply by cloneofghosts** [reply]
> So sorry about that! What happens is that I get focused on something I think is almost done (in this case self-hosting), and think "I'll just reply to the other stuff once that's done". However, when it ends up taking longer than I anticipate, it creates these major issues. On it now!

Didn't you have a reminder setup for it to remind you once a week to check the GH issues? Not sure if you still have it but could be useful in times when you get too focused on something.

**Comment by cloneofghosts** [comment]
Thought about creating a new issue for this but figured since I used this issue in the past. @alexander0042 I've noticed that besides the commits you did yesterday you've gone MIA again. I'm helping where I can but a lot of the times it gets to the point where I'm unable to solve the issue (for whatever reason) and is waiting for you to look at.

I ping you in those cases but it seems like unless you're around or the issue is critical it doesn't seem to make a difference whether you reply or not. I thought you had a reminder setup to look at the GitHub issues at least once per week. Is that no longer the case or you haven't had a chance to look at things?

**Reply by cloneofghosts** [reply]
I've also noticed that you never seem to reply/commit code on the weekends or during the evening. There a reason for that?

**Reply by alexander0042** [reply]
Just let me start by saying your help is deeply appreciated, and I know that me suddenly disappearing makes a already hard task that much harder, So biggest thing here is to just apologize, since at the very (very!) least I could do a way better job of communicating. I don't have any grand explanation or anything- just life with a small baby and grumpy toddler. 

I do read all the GitHub e-mails as they come in, and try to triage anything critical, but there's definitely room for improvement on that front. The other thing that's happened recently is working out some AWS infrastructure kinks with the updated docker ingest files. Doesn't always show up here (although the new building stuff is helps with that), but is a task that needs to be done to keep the data flowing. 

Other than saying that it should get better now that the littlest one is almost sleeping through the night, I think (hope) have one idea that might help. I'm setting up a pathway to let the response docker image read the zarr files directly off my S3, instead of having to run the whole production setup. This is how the weather maps will work, but I'm hoping it might also help people test and debug the response without needing the whole complex setup- if you think it would be helpful? It'll basically be new API endpoint (zarr.pirateweather.net/hrrr?) that the container can be configured to read instead of a local file. I was waiting for a response to [this discussion](https://github.com/zarr-developers/zarr-python/discussions/1613), but it looks like something is there now, so I'm hoping to implement a beta version next week.  

TL;DR: Apologies. Thank you for everything you do to keep this project flowing. Easier community development in the near future?

**Reply by cloneofghosts** [reply]
@alexander0042 So the plan is to have static zarr files in S3 and the API would read directly from those files instead of downloading and processing the gribs locally?

I understand that you're busy with life+work but ideally you'd at least look through GitHub at least once a week to respond to things with any API or urgent issues you'd respond more quickly. There's two main issues that have popped up recently https://github.com/Pirate-Weather/pirateweather/issues/458 and https://github.com/Pirate-Weather/pirateweather/issues/459 that I'm waiting for you to look into. The TimeMachine one you can take your time with as it isn't used as much as the API.

**Reply by alexander0042** [reply]
Yes, totally agree, and I think it'll be better going forward! #458 is priority 1 for Monday since I agree it's an urgent problem.

Re: the zarr files. They'll be the live, updating production ones, so will return the same data as the production/ dev api. Responses will be slower, since accessing zip files on S3 isn't as fast, but still should be within a second or so. In short, it'll let the response container run locally and return the same data as if you were running the whole stack, which is perfect for development.

**Reply by cloneofghosts** [reply]
So basically someone will have the option to host everything themselves or read the data directly off of S3 if they don't have the requirements to host everything themselves?

I guess it would also be good to update https://github.com/Pirate-Weather/pirate-weather-code/issues/85 at the same time as well since the current README is a bit outdated due to the recent changes.

**Reply by alexander0042** [reply]
Yes, exactly. More to the point, it'll let people work on continuing to develop the response container (which is where the interesting stuff is) without having to run the entire stack, which I think will make these sorts of changes easier.
